src/basilisp/contrib/nrepl_server.lpy

```diff
@@ -0,0 +1,476 @@
+           socketserver
+           sys
+           traceback
+           uuid))
+
+(def ^:private logger
+  "The logger for this namespace."
+  (logging/getLogger (namespace ::)))
+
+(defmacro ^:private debug [& values]
+  (when (.isEnabledFor logger logging/DEBUG)
+    `(.debug logger (str/join " " [~@values]))))
+(defmacro ^:private info [& values]
+  (when (.isEnabledFor logger logging/INFO)
+    `(.info logger (str/join " " [~@values]))))
+(defmacro ^:private warn [& values]
+  (when (.isEnabledFor logger logging/WARNING))
+  `(.warning logger (str/join " " [~@values])))
+(defmacro ^:private error [& values]
+  ;; we want error to always output to `sys/stderr` unfiltered.
+  `(.write sys/stderr (str ~(keyword (str *ns*) "error") " " (str/join " " [~@values]) \newline)))
+
+(definterface IReadWrite
+  ;; Pythonic interface for creating `sys/stdout` like File objects.
+  (read [])
+  (write [value]))
+
+(deftype StreamOutFn [out-fn]
+  ;; A type to use as replacement binding for writing to `sys/stdout`
+  ;; stream, so that the output ``value`` is passed to ``out-fn``
+  ;; instead.
+  IReadWrite
+  (read [_this]
+    (throw (python/RuntimeError (str ::logic-error " shouldn't be possible to get here..."))))
+  (write [_self value]
+    (out-fn value)))
+
+(defn- response-for-mw [handler]
+  (fn [{:keys [id session] :as request} response]
+    (let [response (cond-> (assoc response
+                                  "id" id)
+                     session (assoc "session" session))]
+      (handler request response))))
+
+(defn- coerce-request-mw [handler]
+  (fn [request send-fn]
+    (handler (update request :op keyword) send-fn)))
+
+(defn- log-request-mw [handler]
+  (fn [request send-fn]
+    (info :request (dissoc request :client*))
+    (handler request send-fn)))
+
+(defn- log-response-mw [handler]
+  (fn [request response]
+    (info :response response)
+    (handler request response)))
+
+(declare ops)
+
+(defn- handle-describe [request send-fn]
+  (send-fn request
+           {"versions" {"basilisp" (let [version (py->lisp ver/VERSION)]
+                                     (assoc (zipmap ["major" "minor" "incremental"]
+                                                    version)
+                                            "version-string" (str/join "." version)))
+                        "python" (let [version (get (.split sys/version " ") 0)]
+                                   (assoc (zipmap ["major" "minor" "incremental"]
+                                                  (py->lisp (.split version ".")))
+                                          "version-string" (py->lisp sys/version)))}
+            "ops" (zipmap (map name (keys ops)) (repeat {}))
+            "status" ["done"]}))
+
+(defn- format-value [_nrepl-pprint _pprint-options value]
+  (pr-str value))
+
+(defn- send-value [request send-fn v]
+  (let [{:keys [client*]} request
+        {:keys [*2 *3 *e]} @client*
+        [v opts] v
+        ns (:ns opts)]
```

Comment:
I know this is a bit of an annoying request, but would be be possible for you to align all of the `let` bindings in this PR? I personally find it much easier to read aligned let bindings.

I've never found a good way to do this automatically to an entire file besides using Emacs unfortunately.

Reply:
I've aligned the let pairs as much I could.

I'm using a lot of map destructuring forms in `let`s that might be too long or are split into two lines and as such cannot be aligned with the rest of the bindings ...